native/src/routes/Pois.tsx

```diff
@@ -41,6 +42,21 @@ export type PropsType = {
   route: RoutePropType<PoisRouteType>
 }
 
+// Calculate distance for all Feature Locations
+const prepareFeatureLocations = (pois: Array<PoiModel>, userLocation?: number[]): Feature<Point>[] => {
+  return pois
+    .map(poi => {
+      const featureLocation: Feature<Point> = poi.featureLocation as Feature<Point>
```

Comment:
I think we should not need a typecast here if you adjust the type of `poi.featureLocation` to `Feature<Point> | null`, the type annotation is also redundant.
```suggestion
      const featureLocation = poi.featureLocation
```